app/connector/FrontendAuthConnector.scala

```diff
@@ -31,14 +35,43 @@ object GovernmentGatewayEnrolment {
   implicit val fmt = Json.format[GovernmentGatewayEnrolment]
 }
 
+case class InternalUserIdentifier(value: String) extends AnyVal
+
+object InternalUserIdentifier {
+  implicit val reads: Reads[InternalUserIdentifier] = new Reads[InternalUserIdentifier] {
+    override def reads(json: JsValue): JsResult[InternalUserIdentifier] =
+      Try((json \ "internalId").as[String])
+        .map(id => JsSuccess(InternalUserIdentifier(id)))
+        .getOrElse(JsError("Error parsing internal user identifier"))
+  }
+
+  implicit def toString(id: InternalUserIdentifier): String = id.value
+}
+
+case class CoAFEAuthority(twoFactorAuthOtpId: Option[String], enrolmentsUri: Option[String], userDetailsLink: String, internalUserIdentifier: InternalUserIdentifier)
+
 object FrontendAuthConnector extends FrontendAuthConnector with ServicesConfig {
   val serviceUrl = baseUrl("auth")
   lazy val http = WSHttp
 }
 
-trait FrontendAuthConnector extends AuthConnector  {
+trait FrontendAuthConnector extends AuthConnector {
+
+  private case class Authority(twoFactorAuthOtpId: Option[String], enrolmentsUri: Option[String], userDetailsLink: String, idsUri: String)
+
+  private object Authority {
+    implicit val reads: Reads[Authority] =
+      ((JsPath \ "twoFactorAuthOtpId").readNullable[String] and
+        (JsPath \ "enrolments").readNullable[String] and
+        (JsPath \ "userDetailsLink").read[String] and
+        (JsPath \ "ids").read[String]).apply(Authority.apply _)
+  }
 
-  def currentCoAFEAuthority()(implicit hc: HeaderCarrier) = http.GET[CoAFEAuthority](s"$serviceUrl/auth/authority")
+  def currentCoAFEAuthority()(implicit hc: HeaderCarrier) =
+    http.GET[Authority](s"$serviceUrl/auth/authority")
+      .flatMap(authority => http.GET[InternalUserIdentifier](s"$serviceUrl${authority.idsUri}").map(internalUserIdentifier =>
```

Comment:
can the call `http.GET[InternalUserIdentifier](s"$serviceUrl${authority.idsUri}")` be extracted to a def